### PR TITLE
187.schema submodule

### PR DESCRIPTION
**Pull request by singhd789**
## Checklist before requesting a review

- [x] I ran my code
- [x] I tried to make my code readable
- [x] I tried to comment my code
- [ ] I wrote a new test, if applicable
- [x] I wrote new instructions/documentation, if applicable
- [x] I ran pytest and inspected it's output
- [ ] I ran pylint and attempted to implement some of it's feedback

### CODE REVIEW

File: meta.yaml
Comment (ilaflott) [inline meta.yaml:11]: wait are you telling me the problem was the lack of `.git` ??

File: fre/make/gfdlfremake/yamlfre.py
Comment (ilaflott) [inline fre/make/gfdlfremake/yamlfre.py:181]: not sure the `parents` thing is worth it here, `fre/make` is pretty unlikely to ever change in location.

File: fre/yamltools/tests/test_combine_yamls.py
Comment (ilaflott) [inline fre/yamltools/tests/test_combine_yamls.py:84]: similar comment RE `.parents` and, while we may move `gfdl_msd_schemas` idk who knows, might be clearer to just have the spot hard coded. will probably break more clearly than a `os.path` function call

File: fre/yamltools/tests/test_combine_yamls.py
Comment (ilaflott) [inline fre/yamltools/tests/test_combine_yamls.py:137]: see above

File: fre/make/gfdlfremake/schema.json
Comment (ilaflott) [inline fre/make/gfdlfremake/schema.json:1]: bless for you deleting things

File: fre/gfdl_msd_schemas
Comment (ilaflott) [inline fre/gfdl_msd_schemas:1]: the comment i'm going to leave here is likely for a future PR, but is this where we want this? if `fms_yaml_tools` simultaneously lists `gfdl_msd_schemas` in it's submodule list too, and we link up `fre/yamltools` to `fms_yaml_tools` and have `fms_yaml_tools` listed as a conda package req, then might having two `fms_yaml_tools` floating around the conda packages be a recipe for trouble? 

thoughts....

**Review by ilaflott — APPROVED** [review]
I'm going to approve but with the nit of- the `.parents` manipulation in the test scripts look more trouble than it's worth. I'll leave it up to @singhd789 to make the final call on that, though

**Comment by singhd789** [comment]
> I'm going to approve but with the nit of- the `.parents` manipulation in the test scripts look more trouble than it's worth. I'll leave it up to @singhd789 to make the final call on that, though

Hmm, you might be right. I'll look into it a little more before merging.

File: meta.yaml
Comment (singhd789) [reply meta.yaml:11]: It might've been. I was following an example I saw somewhere and they included the `.git`

File: fre/make/gfdlfremake/yamlfre.py
Comment (singhd789) [reply fre/make/gfdlfremake/yamlfre.py:181]: I just saw it as another easy way to define absolute paths / find the directory names ... I'll look into if it can introduce issues

File: fre/gfdl_msd_schemas
Comment (singhd789) [reply fre/gfdl_msd_schemas:1]: Wait, sorry, why would we want to have two `fms_yaml_tools` floating around?

File: fre/yamltools/tests/test_combine_yamls.py
Comment (singhd789) [reply fre/yamltools/tests/test_combine_yamls.py:137]: I see what you mean, especially for tests. Might be better to have that path laid out explicitly

File: fre/make/gfdlfremake/yamlfre.py
Comment (singhd789) [reply fre/make/gfdlfremake/yamlfre.py:181]: I think I'll keep this one here. It finds the right schema location in reference to this file itself, imo, in a simple way

File: fre/gfdl_msd_schemas
Comment (ilaflott) [reply fre/gfdl_msd_schemas:1]: if `gfdl_msd_schemas` is a submodule of `fre-cli` and `fms_yaml_tools`, and `fms_yaml_tools` were a conda package dependency of `fre-cli`, then one would have both `fre/gfdl_msd_schemas` and `fms_yaml_tools/schema/gfdl_msd_schemas` floating around in the environment's `site-packages` directory. At worst, one may worry about an ambiguity arising between the two, or clashing information. At best, it's redundant and not satisfying.

It's not something that needs fixing now- it's a weak spot to jot down for future improvement.

File: meta.yaml
Comment (ilaflott) [reply meta.yaml:11]: OK, if we end up with a mysterious problem at the `publish_conda` step, we'll know @bcc2761's old comment still applies.

**Pull request merged**
